GEOSldas_App/util/postproc/ObsFcstAna_stats/user_config.py

```diff
@@ -0,0 +1,132 @@
+    # If the default "species" number/order do not match, set "species_list" accordingly to force a match.
+    # Output will be cross-masked between all specified experiments.    
+    
+    exp_sup1 = {
+        'expdir'        : '/discover/nobackup/projects/gmao/merra/iau/merra_land/SMAP_runs/SMAP_Nature_v11/',
+        'expid'         : 'DAv8_M36',
+        'exptag'        : 'DASMAP_SMAP', 
+        'domain'        : 'SMAP_EASEv2_M36_GLOBAL',
+        'species_list'  : [1,2,3,4],
+        'obsparam_time' : "20150401_0000"                 # time stamp of obsparam file (YYYYMMDD_HHMM)
+    }
+
+    # Convert experiments input to a list; first entry must be exp_main 
+
+    exp_list = [exp_main, exp_sup1]
+    
+    # Forecasts and analyses are always from the main experiment.
+    # Observations can be from experiment indicated by 'obs_from' index (0-based).
+    # The mostly likely use case for this is that _scaled_ observations from a DA experiment
+    #   are used to compute OmF etc diagnostics for a corresponding open loop experiment.
+
+    obs_from = 1              # 0-based index of exp in exp_list from which obs are extracted
+
+    # Top level directory to store monthly sum files; can use the experiment directory or a different path;
+    # /Yyyyy/Mmm/ is added automatically for individual months
+
+    out_path = '/discover/nobackup/qliu/SMAP_test/'
+    
+    sum_path = out_path+'/'+exp_main['expid']+'/output/'+exp_main['domain']+'/ana/ens_avg/'
+
+    #
+    # ===================== end of user-defined inputs =================================================
+
+    
+    # some minor checks and processing of user-defined inputs
+    
+    if obs_from >= len(exp_list):
+        print('Invalid "obs_from" value')
+        sys.exit()
+
+    # process time range info;  end_time is first of month after (end_year, end_month)
+
+    if last_month==12 :
+        end_month = 1
+        end_year  = last_year + 1
+    else :
+        end_month = last_month + 1
+        end_year  = last_year
+        
+    start_time = datetime( start_year, start_month, 1)
+    end_time   = datetime( end_year,   end_month,   1)
+
+    # Get tilecoord and obsparam information for each experiment
+    
+    for exp in exp_list:
+        expdir   = exp['expdir']
+        expid    = exp['expid']
+        domain   = exp['domain']
+
+        YYYY     = exp['obsparam_time'][0:4]
+        MM       = exp['obsparam_time'][4:6]
+        
+        fop      = expdir+expid+'/output/'+domain+'/rc_out/Y'+YYYY+'/M'+MM+'/'+expid+'.ldas_obsparam.'+exp['obsparam_time']+'z.txt'
+        obsparam = read_obs_param(fop)
+
+        # get the species list and default to list of all species if doesn't exist 
+        species_list = exp.get('species_list',[int(obsparam[i]['species']) for i in range(len(obsparam))])
+        
+        # reorder obsparam to match across experiments
+        obsparam_new = []
+        for i in range(len(obsparam)):
```

Comment:
Maybe here or in a separate loop we can assert that the obsparam info across the experiments does indeed match (detect faulty user input)